cairo/tests/utils/strategies.py

```diff
@@ -447,49 +470,60 @@ def bounded_u256_strategy(min_value: int = 0, max_value: int = 2**256 - 1):
 SYSTEM_ACCOUNT = Account(balance=U256(0), nonce=Uint(0), code=bytes())
 BEACON_ROOTS_ACCOUNT = Account(balance=U256(0), nonce=Uint(1), code=BEACON_ROOTS_CODE)
 
-state = st.lists(address, max_size=MAX_ADDRESS_SET_SIZE, unique=True).flatmap(
-    lambda addresses: st.builds(
-        State,
-        _main_trie=st.builds(
+
+@st.composite
+def state_strategy(draw):
+    addresses = draw(st.lists(address, max_size=MAX_ADDRESS_SET_SIZE, unique=True))
+
+    _main_trie = draw(
```

Comment:
i think you can build this trie after the storage trie is built so that you don't need to update accounts